### ISSUE TITLE
Better Tx Decoder

### ISSUE THREAD

**Issue by hamdiallam** [FourthState/plasma-mvp-sidechain#169]
#33  Summary
The current Tx decoder brute forces decoding the tx with RLP

## Proposal
Have transaction bytes encoded to a json format
```
{
 "type": "<msg name>",
 "tx": "<rlp bytes>"
}
```

**Comment by hamdiallam** [comment]
@colin-axner 

do you think amino is an appropriate use case for this? Or use json directly. 

I know amino effectively does this same thing for de-serializing types structs

**Comment by colin-axner** [comment]
Yea I think you could just wrap the RLP encoded bytes into a struct and register that type on the codec and it should work, but these extra bytes would be included into blocks then, so maybe not?

**Comment by hamdiallam** [comment]
Yeah, I realized this last night. I'll leave this until after the release
